listenbrainz/webserver/views/user.py

```diff
@@ -139,6 +136,28 @@ def profile(user_name):
     )
 
 
+@user_bp.route("/<user_name>/artists")
+def artists(user_name):
+    """ Show the top artists for the user. These users must have been already
+        calculated using Google BigQuery. If the stats are not present, we
+        redirect to the user page with a message.
+    """
+    user = _get_user(user_name)
+    data = db_stats.get_user_artists(user.id)
+
+    # if no data, flash a message and return to profile page
+    if data is None:
+        flash.error('No data calculated for user {user_name} yet.'.format(user_name=user_name))
```

Comment:
I really think we need to improve this error message. If someone sees this, can they expect to wait a few seconds for the data to appear? If not, can they do anything to get the data to appear?